<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml"><head><title>Test_exceptions (qcheck-core.QCheck2.Test_exceptions)</title><meta charset="utf-8"/><link rel="stylesheet" href="../../../odoc.support/odoc.css"/><meta name="generator" content="odoc 3.0.0"/><meta name="viewport" content="width=device-width,initial-scale=1.0"/><script src="../../../odoc.support/highlight.pack.js"></script><script>hljs.initHighlightingOnLoad();</script></head><body class="odoc"><nav class="odoc-nav"><a href="../index.html">Up</a> â€“ <a href="../../../index.html">Index</a> &#x00BB; <a href="../../index.html">qcheck-core</a> &#x00BB; <a href="../index.html">QCheck2</a> &#x00BB; Test_exceptions</nav><header class="odoc-preamble"><h1>Module <code><span>QCheck2.Test_exceptions</span></code></h1></header><div class="odoc-content"><div class="odoc-spec"><div class="spec exception anchored" id="exception-Test_fail"><a href="#exception-Test_fail" class="anchor"></a><code><span><span class="keyword">exception</span> </span><span><span class="exception">Test_fail</span> <span class="keyword">of</span> string * <span>string list</span></span></code></div><div class="spec-doc"><p>Exception raised when a test failed, with the list of counter-examples. <code>Test_fail (name, l)</code> means test <code>name</code> failed on elements of <code>l</code>.</p></div></div><div class="odoc-spec"><div class="spec exception anchored" id="exception-Test_error"><a href="#exception-Test_error" class="anchor"></a><code><span><span class="keyword">exception</span> </span><span><span class="exception">Test_error</span> <span class="keyword">of</span> string * string * exn * string</span></code></div><div class="spec-doc"><p>Exception raised when a test raised an exception <code>e</code>, with the sample that triggered the exception. <code>Test_error (name, i, e, st)</code> means <code>name</code> failed on <code>i</code> with exception <code>e</code>, and <code>st</code> is the stacktrace (if enabled) or an empty string.</p></div></div><div class="odoc-spec"><div class="spec exception anchored" id="exception-Test_unexpected_success"><a href="#exception-Test_unexpected_success" class="anchor"></a><code><span><span class="keyword">exception</span> </span><span><span class="exception">Test_unexpected_success</span> <span class="keyword">of</span> string</span></code></div><div class="spec-doc"><p>Exception raised when a negative test failed. <code>Test_unexpected_success name</code> means test <code>name</code> failed to find an expected counter example.</p></div></div></div></body></html>
